piDIOComm.c

```diff
@@ -151,11 +152,15 @@ INT32U piDIOComm_Init(INT8U i8uDevice_p)
 
 INT32U piDIOComm_sendCyclicTelegram(INT8U i8uDevice_p)
 {
-	INT32U i32uRv_l = 0;
-	SIOGeneric sRequest_l;
-	SIOGeneric sResponse_l;
-	INT8U len_l, data_out[18], i, p, data_in[70];
+	INT8U i, p, data_in[70];
 	INT8U i8uAddress;
+
+	u8 in_buf[IOPROTOCOL_MAXDATA_LENGTH];
+	u8 out_buf[18];
+	u8 snd_buf[18];
+	u16 snd_len;
+	u16 rcv_len;
+	u8 cmd;
 	int ret;
 	static INT8U last_out[40][18];
 #ifdef DEBUG_DEVICE_DIO
```

Comment:
You've left the declaration of `last_in` in place even though you've removed the debug section below which uses that array. This may cause a build warning because of an unused variable declaration if `DEBUG_DEVICE_DIO` is enabled.